src/main/groovy/io/gatling/gradle/GatlingEnterpriseStartTask.groovy

```diff
@@ -9,45 +11,65 @@ import org.gradle.api.tasks.TaskExecutionException
 
 @CacheableTask
 class GatlingEnterpriseStartTask extends DefaultTask {
-
     @Internal
     Closure simulations
 
     @TaskAction
     void publish() {
         def gatling = project.extensions.getByType(GatlingPluginExtension)
-        EnterpriseClient enterpriseClient = gatling.enterprise.getEnterpriseClient(project.version.toString())
 
         def systemProps = gatling.enterprise.systemProps ?: [:]
 
-        if (!gatling.enterprise.simulationId && !gatling.enterprise.simulationClass) {
-            throw new TaskExecutionException(this,
-                new IllegalArgumentException("You need to either configure gatling.enterprise.simulationId if you want to start an existing simulation," +
-                "or gatling.enterprise.simulationClass if you want to create a new simulation")
-            )
-        }
-
         def simulationAndRunSummary
+
         if (gatling.enterprise.simulationId) {
-            simulationAndRunSummary = enterpriseClient.startSimulation(gatling.enterprise.simulationId, systemProps, inputs.files.singleFile)
-        } else {
-            simulationAndRunSummary = enterpriseClient.createAndStartSimulation(
-                gatling.enterprise.teamId,
-                project.group.toString(),
-                project.name,
-                gatling.enterprise.simulationClass,
-                systemProps, inputs.files.singleFile)
+            EnterprisePlugin enterprisePlugin = gatling.enterprise.initEnterprisePlugin(project.version.toString(), logger)
+            simulationAndRunSummary = enterprisePlugin.uploadPackageAndStartSimulation(gatling.enterprise.simulationId, systemProps, inputs.files.singleFile)
+        } else if (gatling.enterprise.batchMode) {
+            if (!gatling.enterprise.simulationId && !gatling.enterprise.simulationClass) {
```

Comment:
Why not split each case implementation in a dedicated method? Easier to read, less depth....